src/gam.py

```diff
@@ -503,6 +503,34 @@ def getCharSet(i):
     return (i, GC_Values.get(GC_CHARSET, GM_Globals[GM_SYS_ENCODING]))
   return (i+2, sys.argv[i+1])
 
+def cleanCourseId(courseId):
+  if courseId.startswith(u'd:'):
+    return courseId[2:]
+  return courseId
+
+def normalizeCourseId(courseId):
+  if not courseId.isdigit() and courseId[:2] != u'd:':
+    return u'd:{0}'.format(courseId)
+  return courseId
+
+def getCourseId(i):
+  if i < len(sys.argv):
+    courseId = sys.argv[i]
+    if courseId:
+      return normalizeCourseId(courseId)
+  print u'ERROR: expected a <CourseId>'
+  sys.exit(2)
+
+def getCourseAlias(i):
+  if i < len(sys.argv):
+    courseAlias = sys.argv[i]
+    if courseAlias:
+      if courseAlias[:2] != u'd:':
+        return u'd:{0}'.format(courseAlias)
+      return courseAlias
+  print u'ERROR: expected a <CourseAlias>'
+  sys.exit(2)
+
```

Comment:
Jay,

I push all of the logic to get an object, courseId in this case, into a
routine. All testing/validation/normalization takes place there, not at the
N call sites.

For example, there are about 30 call sites for Booleans

No common routine:

Typical call site:

    if myarg == u'selected':

      if sys.argv[i+1].lower() in true_values:

        body[u'selected'] = True

      elif sys.argv[i+1].lower() in false_values:

        body[u'selected'] = False

      else:

        print u'ERROR: Value for selected must be true or false; got %s' %
sys.argv[i+1]

        sys.exit(2)

      i += 2

    elif myarg == u'hidden':

      if sys.argv[i+1].lower() in true_values:

        body[u'hidden'] = True

      elif sys.argv[i+1].lower() in false_values:

        body[u'hidden'] = False

      else:

        print u'ERROR: Value for hidden must be true or false; got %s' %
sys.argv[i+1]

        sys.exit(2)

      i += 2


Common routine:

def getBoolean(i, item):

  if i < len(sys.argv):

    boolean = sys.argv[i].strip().lower()

    if boolean in true_values:

      return True

    if boolean in false_values:

      return False

    print u'ERROR: Value for %s must be true or false; got %s' % (item,
sys.argv[i])

    sys.exit(2)

  print u'ERROR: Expected %s <Boolean>' % (item)

  sys.exit(2)


Typical call site:

   if myarg == u'selected':

      body[u'selected'] = getBoolean(i+1, myarg)

      i += 2

    elif myarg == u'hidden':

      body[u'hidden'] = getBoolean(i+1, myarg)

      i += 2

Ross





On Wed, Aug 31, 2016 at 8:44 AM, Jay Lee <notifications@github.com> wrote:

> In src/gam.py
> <https://github.com/jay0lee/GAM/pull/265#discussion_r77013052>:
>
> > +    courseId = sys.argv[i]
> > +    if courseId:
> > +      return normalizeCourseId(courseId)
> > +  print u'ERROR: expected a <CourseId>'
> > +  sys.exit(2)
> > +
> > +def getCourseAlias(i):
> > +  if i < len(sys.argv):
> > +    courseAlias = sys.argv[i]
> > +    if courseAlias:
> > +      if courseAlias[:2] != u'd:':
> > +        return u'd:{0}'.format(courseAlias)
> > +      return courseAlias
> > +  print u'ERROR: expected a <CourseAlias>'
> > +  sys.exit(2)
> > +
>
> These functions seem overly complicated. We should be looking to solve the
> minimal issue (course aliases that are numeric without adding a bunch of
> overhead functions.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/jay0lee/GAM/pull/265/files/7ca254a5cf5eea864d0ce6fcfd12879d79d160ed#r77013052>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AIU8LwJbVaYkTtjxBYQQFCDrf3O9QGKvks5qlaFugaJpZM4JxqTW>
> .
>



-- 
Ross Scroggs
ross.scroggs@gmail.com